### PR TITLE
[SYCL][Joint Matrix] Resolve ambiguous fabs calls, refactor tests to remove some duplication.

### PR DESCRIPTION
**Pull request by YuriPlyakhin**
Main purpose of this PR is to resolve ambiguous fabs calls. While doing this, I also did some refactoring to reduce amount of duplication.

### CODE REVIEW

**Review by dkhaldi — CHANGES_REQUESTED** [review]
Are the changes in Legacy necessary as these use short instead of bfloat16 to represent bfloat16 type?

**Comment by YuriPlyakhin** [comment]
> Are the changes in Legacy necessary as these use short instead of bfloat16 to represent bfloat16 type?

Yes, changes in Legacy folder are also necessary, since tests there are using fabs without namespace as well.
I did not find where do you see usage of short in my changes. make_fp32 which I removed from Legacy tests and replaced with common.hpp seems to be identical.

File: sycl/test-e2e/Matrix/Legacy/joint_matrix_bfloat16_colmajorA_colmajorB.cpp [hunk 1]
```diff
@@ -15,12 +15,10 @@
 
 // XFAIL: gpu
 
-#include <iostream>
```
[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/Legacy/joint_matrix_bfloat16_colmajorA_colmajorB.cpp:18]: resolved

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , all green, please, merge

**Pull request merged**